### PR TITLE
Fix volhist issues

### LINKED ISSUE #451: VolHist: does not warn properly if invalid Voi.
```
>>> dvh = VolHist(d,voi)
/home/bassler/.local/lib/python3.6/site-packages/pytrip/volhist.py:118: RuntimeWarning: invalid value encountered in true_divide
  y = 100.0 * _hist / _hist[0]  # volume histograms always plot the right edge of bin, since V(D < x_pos).
>>> pt.__version__
'2.3.0'
```
Using TST003 from pytrip/tests/res:

```
import pytrip as pt

c = pt.CtxCube()
c.read("tst003000.ctx.gz")

v = pt.VdxCube(c)
v.read("tst003000.vdx")

d = pt.DosCube()
d.read("tst003001.dos.gz")

voi = v.vois[1]

from pytrip.volhist import VolHist
dvh = VolHist(d,voi)
```

### LINKED ISSUE #452: VolHist: does not work with some LetCubes
```
  File "/home/bassler/.local/lib/python3.6/site-packages/pytrip/volhist.py", line 43, in __init__
    self.target_dose = cube.target_dose  # optional target dose scaling factor
AttributeError: 'LETCube' object has no attribute 'target_dose'
Aborted
```
which makes sense when you look at the code. LETCubes may not have the target_dose attribute.

### PR DESCRIPTION
**Pull request by nbassler**
Fix #451 
Fix #452

### CODE REVIEW

File: pytrip/volhist.py [hunk 1]
```diff
@@ -109,6 +115,13 @@ def volume_histogram(data_cube, voi=None, bins=256):
             vcube = voi.get_voi_cube()
             mask = (vcube.cube == 1000)
 
+        if True not in mask:
```
Comment (grzanka) [inline pytrip/volhist.py]: Better `if not mask.any():`

File: pytrip/volhist.py [hunk 2]
```diff
@@ -109,6 +115,13 @@ def volume_histogram(data_cube, voi=None, bins=256):
...
+            warnings.warn("Given VOI has no extend and contains no voxels.",
+                          UserWarning)
+            x = np.arange(100, dtype=np.float64)
+            y = np.zeros(100, dtype=np.float64)
+            return x, y
```
Comment (grzanka) [inline pytrip/volhist.py]: I think we should return here `None,None` and handle it properly on the other (GUI) side

File: pytrip/let.py [hunk 3]
```diff
@@ -49,6 +49,7 @@ class LETCube(Cube):
 
     def __init__(self, cube=None):
         super(LETCube, self).__init__(cube)
+        self.type = "LET"
```
Comment (grzanka) [inline pytrip/let.py]: What is that ?

File: pytrip/volhist.py (hunk 1 above)
Comment (grzanka) [reply pytrip/volhist.py]: Also tests fails with:

```
>       if True not in mask:
E       TypeError: argument of type 'NoneType' is not iterable
```

File: pytrip/let.py (hunk 3 above)
Comment (nbassler) [reply pytrip/let.py]: DosCube has one, so why should LETCube not have one too?

Comment (grzanka) [reply pytrip/let.py]: I forgot to remove it from initial import, it wasn't used if I am correct.

Comment (nbassler) [reply pytrip/let.py]: uh, I intended to use it for the GUI

Comment (grzanka) [reply pytrip/let.py]: why not a class name / type ?

Comment (nbassler) [reply pytrip/let.py]:
```
import pytrip as pt
if isinstance(obj, pytrip.DosCube):
     foobar()
```
vs.
```
if obj.type =='DOS';
    foobar()
```

you can save one line of code, and I just found the latter solution a bit leaner and more convenient, even if less pythonic.

File: pytrip/volhist.py [hunk 4]
```diff
@@ -109,6 +115,11 @@ def volume_histogram(data_cube, voi=None, bins=256):
             vcube = voi.get_voi_cube()
             mask = (vcube.cube == 1000)
 
+        if not mask.any():
```
Comment (grzanka) [inline pytrip/volhist.py]:
```
        if voi is None:
            mask = None
        else:
            vcube = voi.get_voi_cube()
            mask = (vcube.cube == 1000)
    
>       if not mask.any():
E       AttributeError: 'NoneType' object has no attribute 'any'
```

mask can be also None

**Comment by nbassler** [comment]
New issue.

File: pytrip/let.py (hunk 3 above)
Comment (nbassler) [reply pytrip/let.py]: lets put that in a new issue

**Pull request merged**